packages/fx-core/src/plugins/solution/fx-solution/solution.ts

```diff
@@ -1106,6 +1126,9 @@ export class TeamsAppSolution implements Solution {
       ctx.config.get(GLOBAL_CONFIG)?.set(LOCAL_DEBUG_TEAMS_APP_ID, maybeTeamsAppId.value);
     }
 
+    // persistent localSettings.json after local debug.
+    localSettingsProvider.save(ctx.localSettings!);
```

Comment:
Now *localSettings.json* becomes the only single source of truth of all local configurations, so it's not enough to only save it after local debug. If *localDebug* step breaks by any error, local configurations are lost.

Make sure *localSettings* is persisted after both *localdebug* and *postlocaldebug* steps, no matter the step itself is succeeded or failed.

Reply:
Thanks Qianhao. You also remind me of **not to refresh** the previous local debug settings provided by resource plugins when add new resource / capability. I will handle these parts later.